asllib/ASTUtils.ml

```diff
@@ -49,17 +49,23 @@ module IMap = struct
 end
 
 let dummy_pos = Lexing.dummy_pos
+let default_version = V1
 let desc v = v.desc
-let annotated desc pos_start pos_end = { desc; pos_start; pos_end }
-let add_dummy_pos desc = annotated desc dummy_pos dummy_pos
-let dummy_annotated = add_dummy_pos ()
+
+let annotated desc pos_start pos_end version =
+  { desc; pos_start; pos_end; version }
+
+let add_dummy_annotation ?(version = default_version) desc =
+  annotated desc dummy_pos dummy_pos version
+
+let dummy_annotated = add_dummy_annotation ()
 let to_pos pos = { pos with desc = () }
 
 let add_pos_from_st pos desc =
   if pos.desc == desc then pos else { pos with desc }
 
 let add_pos_from pos desc = { pos with desc }
-let with_pos_from pos { desc; _ } = add_pos_from pos desc
+let with_pos_from pos { desc; version } = add_pos_from { pos with version } desc
```

Comment:
My thinking was that if given an AST generated by the V0 parser, we don't want to accidentally consider this as V1 AST, as it may become incompatible with V1 checking logic. Looking more closely, I realise that `with_pos_from` is used only once (in `StaticModel.ml`) - perhaps it would be better if I inlined its previous definition there?

Reply:
Also forgot to say - for the compiler not giving warnings, from [what I can see](https://ocaml.org/manual/5.2/patterns.html#sss:pat-record) it doesn't mind for records:
> The record value can define more fields than field1 … fieldn; the values associated to these extra fields are not taken into account for matching.